lib/src/config_parser.cc

```diff
@@ -254,7 +285,15 @@ namespace hocon { namespace config_parser {
                         throw new bug_or_broken_exception("somehow got multi-element path in JSON mode");
                     }
 
-                    throw bug_or_broken_exception("createValueUnderPath not implemented");
+                    shared_object obj = create_value_under_path(remaining, new_value);
+                    auto existing = values.find(*key);
+                    if (existing != values.end()) {
+                        obj = dynamic_pointer_cast<const config_object>(obj->with_fallback(existing->second));
+                        assert(obj);
+                        existing->second = obj;
+                    } else {
```

Comment:
That confused me too, but doing it just like the java was actually a bug. For some reason the value was not replacing properly from unordered_map::insert, so instead I have two cases: update the object already stored in the map, or add a new object to the map. I copied an example of this I found elsewhere in that file.